controllers/controllers.go

```diff
@@ -16,7 +16,49 @@
 
 package controllers
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/openshift-kni/numaresources-operator/pkg/status"
+)
+
+type conditionInfo struct {
+	Type    string // like metav1.Condition.Type
+	Reason  string
+	Message string
+}
+
+func availableConditionInfo() conditionInfo {
+	return conditionInfo{
+		Type:   status.ConditionAvailable,
+		Reason: "AsExpected",
+	}
+}
+
+func progressingConditionInfo(reason, message string) conditionInfo {
+	return conditionInfo{
+		Type:    status.ConditionProgressing,
+		Message: message,
+		Reason:  reason,
+	}
+}
+
+func degradedConditionInfoFromError(err error) conditionInfo {
+	return conditionInfo{
+		Type:    status.ConditionDegraded,
+		Message: messageFromError(err),
+	}
+}
+
+func fillConditionInfoFromError(cond conditionInfo, err error) conditionInfo {
+	if cond.Message == "" {
+		cond.Message = messageFromError(err)
+	}
+	if cond.Reason == "" {
+		cond.Reason = reasonFromError(err)
+	}
+	return cond
```

Comment:
the only path the reconciliation will return an error is when the condition is degraded, do we still need this function or can we just integrate the reason in degradedConditionInfoFromError?

Reply:
seems unnecessary indeed, removed